[WFLY-17961] Spurious errors from Micrometer

jasondlee opened the pull request:
https://issues.redhat.com/browse/WFLY-17961

Implement a no-op MeterRegistry
Make the endpoint not required
If the endpoint is not specified, register the no-op MeterRegistry

bstansberry commented on observability/micrometer/src/main/java/org/wildfly/extension/micrometer/MicrometerExtensionLogger.java:
```diff
@@ -75,4 +75,8 @@ public interface MicrometerExtensionLogger extends BasicLogger {
     void notActivatingSubsystem();
+
+    @LogMessage(level = WARN)
+    @Message(id = 11, value = "Micrometer has been enabled, but no endpoint has been configured. A No-op registry has been configured.")
```
s/registry/metrics registry/g ?

jasondlee replied:
Changed.

bstansberry commented on observability/micrometer/src/main/java/org/wildfly/extension/micrometer/MicrometerRegistryService.java:
```diff
@@ -18,19 +18,22 @@
+
+import java.io.IOException;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
```
These were in the right spot before :)

jasondlee replied:
Grr. I had IDEA configured correctly at one point. :P

bstansberry commented on observability/micrometer/src/main/java/org/wildfly/extension/micrometer/MicrometerSubsystemDefinition.java:
```diff
@@ -68,7 +68,7 @@ class MicrometerSubsystemDefinition extends PersistentResourceDefinition {
             .create(MicrometerConfigurationConstants.ENDPOINT, ModelType.STRING)
             .setAttributeGroup(MicrometerConfigurationConstants.OTLP_REGISTRY)
-            .setRequired(true)
+            .setRequired(false)
```
You need to bump the management API version for this. XSD as well.

bstansberry replied:
Note that this subsystem is not supported in domain mode in WF 28, so there is no need to deal with mixed domain transformation.

darranl reviewed (changes requested):
Please separate the model bump aspect into it's own commit.  Using current tooling it is not possible to see what has changed in the schema and in future using git annotate it will not be possible to see the first changes that were made to the schema as it all shows as a new file in a single commit.

jasondlee committed:
[WFLY-17961] Spurious errors from Micrometer

Bump model and schema versions to 1.1

jasondlee committed:
[WFLY-17961] Spurious errors from Micrometer

Implement a no-op MeterRegistry
Make the endpoint not required
If the endpoint is not specified, register the no-op MeterRegistry

jasondlee commented:
/retest

darranl commented:
/retest

darranl commented:
Just removing ready for merge as it is suspected of being related to a test failure on the aggregate job.

darranl commented:
Sorry @jasondlee this one has a merge conflict, also it is the Bootable Jar Linux run we saw failing in the aggregate job which was why this one didn't go ahead.

jasondlee commented:
> Sorry @jasondlee this one has a merge conflict, also it is the Bootable Jar Linux run we saw failing in the aggregate job which was why this one didn't go ahead.

Not a problem. i'll get that resolved ASAP.
